src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -168,6 +168,62 @@ public RTPEncodingImpl(
     }
 
     /**
+     * Applies frame boundaries heuristics to frames a and b, assuming a
+     * predates/is older than b.
+     *
+     * @param a the old {@link FrameDesc}.
+     * @param b the new {@link FrameDesc}
+     */
+    private static void applyFrameBoundsHeuristics(FrameDesc a, FrameDesc b)
+    {
+        int end = a.getEnd(), start = b.getStart();
+        if (end != -1 && start != -1)
+        {
+            // No need for heuristics.
+            return;
+        }
+
+        long tsDiff = (b.getTimestamp() - a.getTimestamp()) & 0xFFFFFFFFL;
+        if (tsDiff > (1L << 30) && tsDiff < (-(1L << 30) & 0xFFFFFFFFL))
+        {
+            // the distance (mod 32) between the two timestamps needs to be
+            // less than half the timestamp space.
+            return;
+        }
+
+        int min = b.getMinSeen(), max = a.getMaxSeen();
+        int snDiff = (max - min) & 0xFFFF;
+
+        if (start != -1 || end != -1)
+        {
+            // We can deal with sequence number distances (mod 16) that are
+            // less than 3.
+            if (snDiff < 3 || snDiff > (-3 & 0xFFFF))
```

Comment:
Same here, if snDiff > (-3 & 0xFFFF), then our frames are in the wrong order, and we shouldn't apply the same logic.

Also, we can apply this only when diff == 2. If diff <2, then there is something wrong (e.g. corrupted stream).